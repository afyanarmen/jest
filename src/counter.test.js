import React from 'react';
import Counter from './counter';
import renderer from 'react-test-renderer';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

describe('Counter component', ()=>{
  test('matches snapshot', ()=>{
    const component = renderer.create(<Counter />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('clicking the button increases the counter', ()=>{
    const component = shallow(<Counter />);
    const prevValue = component.find('p').text();
    component.find('button').simulate('click');
    const currValue = component.find('p').text();
    expect(currValue - prevValue).toEqual(1);
  });
});