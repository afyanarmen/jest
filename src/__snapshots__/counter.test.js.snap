// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component matches snapshot 1`] = `
<div>
  <p>
    0
  </p>
  <button
    onClick={[Function]}
  >
    Click
  </button>
</div>
`;
